.github/workflows/partial-tests.yml

```diff
@@ -0,0 +1,106 @@
+name: Partial Tests Workflow
+
+on:
+  push:
+    branches: [ develop, dev/*, release/* ]
+  pull_request:
+    branches: [ develop, dev/*, release/* ]
+
+jobs:
+  linter:
+    runs-on: ubuntu-latest
+    steps:
+      # replace for Ruff in the future
+      - uses: actions/checkout@v4
+      - uses: ricardochaves/python-lint@v1.4.0
+        with:
+          use-pylint: false
+          use-isort: false
+          use-mypy: false
+          extra-black-options: "--line-length=120 --diff"
+          extra-pycodestyle-options: "--max-line-length=120  --exclude=tests/gui --ignore=E121,E123,E126,E226,E24,E704,W503,W504,E203"
+          extra-mypy-options: "--ignore-missing-imports --implicit-optional --no-namespace-packages --exclude (taipy/templates/|generate_pyi.py) --follow-imports skip"
+          extra-isort-options: "--line-length=120 --force-grid-wrap=10 --multi-line=VERTICAL_HANGING_INDENT --trailing-comma"
+  tests:
+    needs: linter
+    timeout-minutes: 40
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ['3.8', '3.9', '3.10', '3.11']
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: dorny/paths-filter@v2
+        id: changes
+        with:
+          filters: |
+            cli:
+              - 'taipy/_cli/**'
+            config:
+              - 'taipy/config/**'
+            core:
+              - 'taipy/core/**'
+            gui:
+              - 'taipy/gui/**'
+            gui-core:
+              - 'taipy/gui_core/**'
+            logger:
+              - 'taipy/logger/**'
+            rest:
+              - 'taipy/rest/**'
```

Comment:
I would add `templates` here as well

Reply:
I didn't add templates because the templates test is still inside the package, it needs to be moved to the root test folder. I can add but effectively will only work after the tests are moved to the correct folder.

Reply:
PR https://github.com/Avaiga/taipy/pull/583 from @toan-quach has already did that, however we did not merge it because of the timeout.
We need to align on this topic